### ISSUE TITLE
X2GO Crash/Java issues

### ISSUE THREAD

**Issue by perara** [ArcticaProject/nx-libs#528]
I'm experiences relatively big issues with running appkications like IntelliJ and CLion in X2GO. These are JVM powered applications.

Issues especially arises ehen opening dialogs in the application. It results in the X2GO session closing (crash). I'm guessing these problems happens ro most java applications. I sadly does not have any logs, but can supply these with some directions.
I have tested and can confirn that  this happens in ubuntu 16.04, 16.10, 17.04 and 17.10. On both stable and devel version of X2GO

**Comment by uli42** [comment]
Are you using openjdk or Oracle java? You can check .java in your home for logfiles, not sure if that has to be activated before (with jcontrol).

Uli

Vom Smartphone gesendet.

----- Ursprüngliche Nachricht -----
Von: "Per-Arne Andersen" <notifications@github.com>
Gesendet: ‎12.‎10.‎2017 15:31
An: "ArcticaProject/nx-libs" <nx-libs@noreply.github.com>
Cc: "Subscribed" <subscribed@noreply.github.com>
Betreff: [ArcticaProject/nx-libs] X2GO Crash/Java issues (#528)

I'm experiences relatively big issues with running appkications like IntelliJ and CLion in X2GO. These are JVM powered applications.
Issues especially arises ehen opening dialogs in the application. It results in the X2GO session closing (crash). I'm guessing these problems happens ro most java applications. I sadly does not have any logs, but can supply these with some directions.
I have tested and can confirn that this happens in ubuntu 16.04, 16.10, 17.04 and 17.10. On both stable and devel version of X2GO

**Comment by perara** [comment]
I'm using oracle. Ive tried both java 8 and 9.
Ill check if i can get any logs from ~/.java

**Comment by perara** [comment]
From ``` /home/<USER>/.cache/lxsession/LXDE/run.log```
```
(lxpanel:11207): Wnck-WARNING **: Unhandled action type _OB_WM_ACTION_UNDECORATE

(lxpanel:11207): Wnck-WARNING **: Unhandled action type _OB_WM_ACTION_UNDECORATE

(lxpanel:11207): Wnck-WARNING **: Unhandled action type _OB_WM_ACTION_UNDECORATE

(nm-applet:11231): Gtk-CRITICAL **: gtk_widget_destroy: assertion 'GTK_IS_WIDGET (widget)' failed

(nm-applet:11231): Gtk-CRITICAL **: gtk_widget_destroy: assertion 'GTK_IS_WIDGET (widget)' failed

(nm-applet:11231): Gtk-WARNING **: Can't set a parent on widget which has a parent
```

Can any of these errors possibly trigger X2GO to crash?

**Comment by Ionic** [comment]
No, these messages are mostly harmless and coming from other applications (lxpanel, that is part of LXDE and nm-applet, which is part of NetworkManager.)

**Comment by perara** [comment]
Could this be client side?

**Comment by perara** [comment]
Which logs could identify a X2GO crash?
I must mention that its not a complete server crash, only a sudden disconnect. In 70% of the cases, i can reconnect to the session, the rest 30% its gone.

**Comment by Hybrid512** [comment]
I encounter same issue with Eclipse mostly with OpenGL enabled or disabled, same behaviour, unexpected crashes of the application but apparently, for my concern, this only occures in Published App mode.

**Comment by uli42** [comment]
Just to be sure: You are talking about "X2go devel version" here. Are you using the nx-libs 3.5.0 or 3.5.99.something?

**Comment by Hybrid512** [comment]
yeap, nx-libs 3.6 branch (3.5.99.10 currently)

**Comment by olivergg** [comment]
This is a known issue. See https://bugs.x2go.org/cgi-bin/bugreport.cgi?bug=645
